File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    participant.sort()
    completion.sort()
    
    # 참가자와 완주자 정렬 후 차이가 나는 첫 번째 이름을 반환
    for i in range(len(completion)):
        if participant[i] != completion[i]:
            return participant[i]  
    
    # 마지막 참가자가 완주하지 못한 경우
    return participant[-1]

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    const participantCount = new Map();
    
    for(let name of participant) {
        const count = participantCount.get(name) || 0;
        participantCount.set(name, count + 1);
    }
    
    for(let name of completion) {
        participantCount.set(name, participantCount.get(name) - 1);
    }
    
    for(let [name, count] of participantCount) {
        if(count >= 1) {
            return name;
        }
    }
    
    return '';
}